packages/core/src/Renderer.ts

```diff
@@ -1,6 +1,25 @@
 import { VDocument } from './VDocument';
+import { VNode } from './VNodes/VNode';
 
-export class Renderer {
-    static readonly id: string;
-    render: (vDocument: VDocument, target: Element) => void;
+export type RenderingMap = Map<VNode, Node[]>;
+export interface RenderingContext {
+    currentVNode?: VNode; // Current VNode rendered at this step.
+    parentNode?: Node | DocumentFragment; // Node to render the VNode into.
+}
+export type RenderingPredicate<T> = (node: VNode) => RenderingFunction<T>;
+export type RenderingFunction<T> = (node: VNode, defaultRendering?: T) => T;
+
+export class Renderer<T> {
+    renderingPredicates = new Set<RenderingPredicate<T>>();
+    readonly name: string;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function, @typescript-eslint/no-explicit-any
```

Comment:
// dmo-disable-next-line @dmo-eslint/no-eslint-disable

Reply:
Well. Agreed. But do you know how I can do without?